File: src/main/java/pm/n2/tangerine/gui/renderables/DemoWindow.java
package pm.n2.tangerine.gui.renderables;

import imgui.ImGui;
import imgui.extension.implot.ImPlot;
import imgui.type.ImBoolean;
import pm.n2.tangerine.gui.TangerineRenderable;

public class DemoWindow extends TangerineRenderable {
	private ImBoolean drawImGuiDemo = new ImBoolean(false);
	private ImBoolean drawImPlotDemo = new ImBoolean(false);

	public DemoWindow() {
		super("DemoWindow", false);
	}

	@Override
	public void draw() {
		var enabled = new ImBoolean(this.enabled);
		if (ImGui.begin("Demo windows", enabled)) {
			ImGui.checkbox("Draw ImGui demo", drawImGuiDemo);
			ImGui.checkbox("Draw ImPlot demo", drawImPlotDemo);
		}

		ImGui.end();
		this.enabled = enabled.get();

		if (drawImGuiDemo.get()) ImGui.showDemoWindow();
		if (drawImPlotDemo.get()) ImPlot.showDemoWindow(drawImPlotDemo);
	}
}

File: src/main/kotlin/pm/n2/tangerine/gui/renderables/DemoWindow.kt
package pm.n2.tangerine.gui.renderables

import imgui.ImGui
import imgui.extension.implot.ImPlot
import imgui.type.ImBoolean
import net.minecraft.client.resource.language.I18n
import pm.n2.tangerine.gui.TangerineRenderable

object DemoWindow : TangerineRenderable("DemoWindow", false) {
    private var drawImGuiDemo = ImBoolean(false)
    private var drawImPlotDemo = ImBoolean(false)

    override fun draw() {
        val enabled = ImBoolean(this.enabled)

        if (ImGui.begin(I18n.translate("tangerine.ui.demo.name"), enabled)) {
            ImGui.checkbox(I18n.translate("tangerine.ui.demo.imgui"), drawImGuiDemo)
            ImGui.checkbox(I18n.translate("tangerine.ui.demo.implot"), drawImPlotDemo)
        }

        ImGui.end()
        this.enabled = enabled.get()

        ImGui.showDemoWindow(drawImGuiDemo)
        ImPlot.showDemoWindow(drawImPlotDemo)
    }
}
